### PR TITLE
API calls for getting all versions of entities

### PR DESCRIPTION
**Pull request by moritzmeister**

### CODE REVIEW

File: java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java [hunk 1]
```diff
@@ -49,6 +50,31 @@
...
+        .expand();
+
+    LOGGER.info("Sending metadata request: " + uri);
+    TrainingDataset[] trainingDatasets = hopsworksClient.handleRequest(new HttpGet(uri), TrainingDataset[].class);
+
+    for (TrainingDataset td : trainingDatasets) {
+      td.setFeatureStore(featureStore);
+      td.getFeatures().stream()
+          .filter(f -> f.getFeaturegroup() != null)
+          .forEach(f -> f.getFeaturegroup().setFeatureStore(featureStore));
```
Comment (SirOibaf) [inline java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java:73]: I'm confused on why we need this - I see that the other get has it as well. It doesn't look correct though - what happens if you are joining features coming from different feature groups?

Comment (SirOibaf) [reply java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java:73]: Also I think we should be DRY here and apply the same logic as we do for the feature groups, so we don't have twice the same get method.

**Review by SirOibaf — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java (hunk 1 above)
Comment (moritzmeister) [reply java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java:73]: The setting of the feature store object on the feature group was introduced when we added the OnlineFS ingestion, I think it is needed to get the schema for complex features when decoding them in the `get_serving_vector()`. I am not sure why we don't have that in the Python API though.
We are only setting the feature store, not the feature group itself, all features in a training dataset should be coming from the same feature store right?

Comment (moritzmeister) [reply java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java:73]: I agree that I can make it dry

Comment (moritzmeister) [reply java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java:73]: okay in python we have set up the API classes a bit differently and we keep a reference to the feature store there, that's why we don't do it there.

**Comment by moritzmeister** [comment]
#473 remember this for future. We set the same feature store for all features in a training dataset. In python we cache it in the API classes which is also incorrect.

File: java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java (hunk 1 above)
Comment (moritzmeister) [reply java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java:77]: see #473

**Review by SirOibaf — APPROVED** [review]

**Pull request merged**